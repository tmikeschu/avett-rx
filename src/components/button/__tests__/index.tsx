import * as React from "react";

import * as utils from "test";

import Button, { BUTTON_COLORS, SIZES, VARIANTS } from "../";

describe("<Button />", () => {
  it("matches snapshot", () => {
    const { getByRole, rerender } = utils.render(<Button>Snapshot</Button>);
    const button = getByRole("button");
    expect(button).toMatchInlineSnapshot(`
      <button
        class="text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-4 py-2 bg-primary text-light border-primary"
      >
        <span
          class="text-dark font-bold text-base sm:text-lg font-body uppercase"
        >
          Snapshot
        </span>
      </button>
    `);

    rerender(<Button disabled={true}>Snapshot</Button>);
    expect(button).toMatchInlineSnapshot(`
      <button
        class="text-base cursor-not-allowed opacity-25 rounded px-4 py-2 bg-primary text-light border-primary"
        disabled=""
      >
        <span
          class="text-dark font-bold text-base sm:text-lg font-body uppercase"
        >
          Snapshot
        </span>
      </button>
    `);

    rerender(
      <Button disabled={true}>
        <pre>Snapshot</pre>
      </Button>
    );
    expect(button).toMatchInlineSnapshot(`
      <button
        class="text-base cursor-not-allowed opacity-25 rounded px-4 py-2 bg-primary text-light border-primary"
        disabled=""
      >
        <pre>
          Snapshot
        </pre>
      </button>
    `);
  });

  describe("variants", () => {
    VARIANTS.forEach((variant) => {
      BUTTON_COLORS.forEach((color) => {
        SIZES.forEach((size) => {
          it(`Variant: ${variant}, color: ${color}, size: ${size} matches snapshot`, () => {
            const { getByRole } = utils.render(
              <Button variant={variant} color={color} size={size}>
                Snapshot
              </Button>
            );
            const button = getByRole("button");
            expect(button.className).toMatchSnapshot();
          });
        });
      });
    });
  });
});
