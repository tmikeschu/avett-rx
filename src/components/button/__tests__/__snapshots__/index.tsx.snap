// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> variants Variant: link, color: cancel, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-gray-600 text-gray-600"`;

exports[`<Button /> variants Variant: link, color: cancel, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-gray-600 text-gray-600"`;

exports[`<Button /> variants Variant: link, color: cancel, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-gray-600 text-gray-600"`;

exports[`<Button /> variants Variant: link, color: error, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-error text-error"`;

exports[`<Button /> variants Variant: link, color: error, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-error text-error"`;

exports[`<Button /> variants Variant: link, color: error, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-error text-error"`;

exports[`<Button /> variants Variant: link, color: primary, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 text-primary border-primary"`;

exports[`<Button /> variants Variant: link, color: primary, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 text-primary border-primary"`;

exports[`<Button /> variants Variant: link, color: primary, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 text-primary border-primary"`;

exports[`<Button /> variants Variant: link, color: secondary, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 text-secondary border-secondary"`;

exports[`<Button /> variants Variant: link, color: secondary, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 text-secondary border-secondary"`;

exports[`<Button /> variants Variant: link, color: secondary, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 text-secondary border-secondary"`;

exports[`<Button /> variants Variant: link, color: success, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-success text-success"`;

exports[`<Button /> variants Variant: link, color: success, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-success text-success"`;

exports[`<Button /> variants Variant: link, color: success, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-success text-success"`;

exports[`<Button /> variants Variant: link, color: warning, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-warning text-black"`;

exports[`<Button /> variants Variant: link, color: warning, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-warning text-black"`;

exports[`<Button /> variants Variant: link, color: warning, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none border-b border-solid bg-none p-0 border-warning text-black"`;

exports[`<Button /> variants Variant: outline, color: cancel, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-8 py-4 border-gray-600 text-gray-600"`;

exports[`<Button /> variants Variant: outline, color: cancel, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-4 py-2 border-gray-600 text-gray-600"`;

exports[`<Button /> variants Variant: outline, color: cancel, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-2 py-1 border-gray-600 text-gray-600"`;

exports[`<Button /> variants Variant: outline, color: error, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-8 py-4 border-error text-error"`;

exports[`<Button /> variants Variant: outline, color: error, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-4 py-2 border-error text-error"`;

exports[`<Button /> variants Variant: outline, color: error, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-2 py-1 border-error text-error"`;

exports[`<Button /> variants Variant: outline, color: primary, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-8 py-4 text-primary border-primary"`;

exports[`<Button /> variants Variant: outline, color: primary, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-4 py-2 text-primary border-primary"`;

exports[`<Button /> variants Variant: outline, color: primary, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-2 py-1 text-primary border-primary"`;

exports[`<Button /> variants Variant: outline, color: secondary, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-8 py-4 text-secondary border-secondary"`;

exports[`<Button /> variants Variant: outline, color: secondary, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-4 py-2 text-secondary border-secondary"`;

exports[`<Button /> variants Variant: outline, color: secondary, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-2 py-1 text-secondary border-secondary"`;

exports[`<Button /> variants Variant: outline, color: success, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-8 py-4 border-success text-success"`;

exports[`<Button /> variants Variant: outline, color: success, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-4 py-2 border-success text-success"`;

exports[`<Button /> variants Variant: outline, color: success, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-2 py-1 border-success text-success"`;

exports[`<Button /> variants Variant: outline, color: warning, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-8 py-4 border-warning text-black"`;

exports[`<Button /> variants Variant: outline, color: warning, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-4 py-2 border-warning text-black"`;

exports[`<Button /> variants Variant: outline, color: warning, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded border border-solid bg-light px-2 py-1 border-warning text-black"`;

exports[`<Button /> variants Variant: solid, color: cancel, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-8 py-4 bg-gray-600 border-gray-600 text-light"`;

exports[`<Button /> variants Variant: solid, color: cancel, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-4 py-2 bg-gray-600 border-gray-600 text-light"`;

exports[`<Button /> variants Variant: solid, color: cancel, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-2 py-1 bg-gray-600 border-gray-600 text-light"`;

exports[`<Button /> variants Variant: solid, color: error, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-8 py-4 bg-error text-light border-error"`;

exports[`<Button /> variants Variant: solid, color: error, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-4 py-2 bg-error text-light border-error"`;

exports[`<Button /> variants Variant: solid, color: error, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-2 py-1 bg-error text-light border-error"`;

exports[`<Button /> variants Variant: solid, color: primary, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-8 py-4 bg-primary text-light border-primary"`;

exports[`<Button /> variants Variant: solid, color: primary, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-4 py-2 bg-primary text-light border-primary"`;

exports[`<Button /> variants Variant: solid, color: primary, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-2 py-1 bg-primary text-light border-primary"`;

exports[`<Button /> variants Variant: solid, color: secondary, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-8 py-4 bg-secondary border-secondary text-light"`;

exports[`<Button /> variants Variant: solid, color: secondary, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-4 py-2 bg-secondary border-secondary text-light"`;

exports[`<Button /> variants Variant: solid, color: secondary, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-2 py-1 bg-secondary border-secondary text-light"`;

exports[`<Button /> variants Variant: solid, color: success, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-8 py-4 bg-success text-light border-success"`;

exports[`<Button /> variants Variant: solid, color: success, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-4 py-2 bg-success text-light border-success"`;

exports[`<Button /> variants Variant: solid, color: success, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-2 py-1 bg-success text-light border-success"`;

exports[`<Button /> variants Variant: solid, color: warning, size: lg matches snapshot 1`] = `"text-lg hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-8 py-4 bg-warning text-dark border-warning"`;

exports[`<Button /> variants Variant: solid, color: warning, size: md matches snapshot 1`] = `"text-base hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-4 py-2 bg-warning text-dark border-warning"`;

exports[`<Button /> variants Variant: solid, color: warning, size: sm matches snapshot 1`] = `"text-sm hover:opacity-75 focus:opacity-75 focus:outline-none rounded px-2 py-1 bg-warning text-dark border-warning"`;
